table/server/master/src/main/java/alluxio/master/table/Table.java

```diff
@@ -87,6 +90,50 @@ public void sync(UdbTable udbTable) {
     }
   }
 
+  private boolean checkSyncable(UdbTable udbTable) {
+    if (Objects.equals(mSchema, udbTable.getSchema())) {
+      // can't sync if the schema is different
+      return false;
+    }
+    if (mPartitionScheme == null || mPartitionScheme.getPartitionCols().isEmpty()) {
+      // can't sync if it is non-partitioned table
+      return false;
+    }
+    List<FieldSchema> partitionCols = new ArrayList<>(udbTable.getPartitionCols());
+    return Objects.equals(partitionCols, mPartitionScheme.getPartitionCols());
+  }
+
+  /**
+   * incremental sync the table with a udbtable.
+   *
+   * @param udbTable udb table to be synced
+   * @return true if the table changed
+   */
+  public boolean incrementalSync(UdbTable udbTable) {
```

Comment:
Is there not a way for just to have a single `sync()` method which handles all the sync cases?